onnxruntime/core/optimizer/insert_cast_transformer.cc

```diff
@@ -231,37 +231,29 @@
   return Status::OK();
 }
 
-enum TypeGroup {
-  Unknown = -1,
-  Bool = 0,
-  Integer = 1,
-  Float = 2,
-};
-
-TypeGroup GetTypeGroup(DataType type) {
-  if (*type == "tensor(bool)") {
-    return Bool;
-  }
-
-  if (*type == "tensor(int16)" || *type == "tensor(int32)" || *type == "tensor(int64)" || *type == "tensor(int8)" ||
-      *type == "tensor(uint16)" || *type == "tensor(uint32)" || *type == "tensor(uint64)" || *type == "tensor(uint8)") {
-    return Integer;
-  }
-
-  if (*type == "tensor(bfloat16)" || *type == "tensor(double)" || *type == "tensor(float)" || *type == "tensor(float16)") {
-    return Float;
-  }
-
-  return Unknown;
-}
-
 /** Transformer to remove duplicate Cast nodes. */
 class RemoveDuplicateCastTransformer : public GraphTransformer {
  public:
   RemoveDuplicateCastTransformer() : GraphTransformer("RemoveDuplicateCastTransformer") {
   }
 
  private:
+  InlinedVector<std::string> cast_ordering{
```

Comment:
>InlinedVector<std::string> cast_ordering [](http://example.com/codeflow?start=2&length=40)

Suggest `constexpr std::array<std::string_view>`

Reply:
Are we allowed to use c++ 17 features in onnxruntime?

Reply:
Yes, Onnxruntime is built with C++17 compiler, it is a requirement. However, we do not impose the same on the public headers.

Reply:
Got it, thanks!